### DISCUSSION TITLE
Auto-create first warehouse

### DISCUSSION THREAD

**Discussion by bwlove319** [theory/sqitch#615] [Ideas]
A nice-to-have feature would be the auto-creation of the requested Snowflake warehouse.

As a user of Sqitch I would like to execute Sqitch on a newly created Snowflake account without having to create a warehouse manually first.

Currently new accounts have an ACCOUNTADMIN user, but no warehouses. This poses a chicken/egg problem, where I want Sqitch to create warehouses, but it can't do that until a warehouse exists.  
Maybe add a "-w" flag or "--create-warehouse" to the command line that would trigger the execution of the following commands before anything else:

CREATE WAREHOUSE if not exists &warehouse WITH WAREHOUSE_SIZE = 'XSMALL' WAREHOUSE_TYPE = 'STANDARD' AUTO_SUSPEND = 30 AUTO_RESUME = TRUE;
USE WAREHOUSE &warehouse;

**Comment by theory** [comment]
The Sqitch user may not have there permissions to create it. On connection it resumes and uses the warehouse:

https://github.com/sqitchers/sqitch/blob/9074a947ce7510a6448edb1056a09c8dd8d36fd3/lib/App/Sqitch/Engine/snowflake.pm#L213-L214

**Comment by bwlove319** [comment]
Right now there is no way to execute Sqitch without first creating a warehouse in a new account.  I think it would improve the usability of Sqitch if the option at least existed.  
If permissions are a problem then the user can solve that however they wish, but they won't be any worse off for it.

The idea is to automate as much of the DB creation as possible.  This mod enables DB creation without a user ever logging in to a new account manually.
And while I could wrap sqitch in a shell script that pre-executes the aforementioned statements, it would be nice to have it built in.
Either way, no big deal.

**Comment by theory** [comment]
Yeah, early on I made the decision that sqitch would not create databases. Too many people wrote in for my other projects to complain that there were errors because they already had databases and limited permissions.
